exporter/signalfxexporter/correlation/correlation_test.go

```diff
@@ -33,20 +37,66 @@ func TestTrackerAddSpans(t *testing.T) {
 		},
 	)
 
+	err := tracker.Start(context.Background(), componenttest.NewNopHost())
+	require.NoError(t, err)
+	assert.NotNil(t, tracker.correlation, "correlation context should be set")
+
 	traces := pdata.NewTraces()
 	rs := traces.ResourceSpans().AppendEmpty()
 	attr := rs.Resource().Attributes()
 	attr.InsertString("host.name", "localhost")
 
 	// Add empty first, should ignore.
 	tracker.AddSpans(context.Background(), pdata.NewTraces())
-	assert.Nil(t, tracker.correlation)
 	assert.Nil(t, tracker.traceTracker)
 
 	tracker.AddSpans(context.Background(), traces)
 
-	assert.NotNil(t, tracker.correlation, "correlation context should be set")
```

Comment:
Why is this line removed?

Reply:
It was moved up and now lives on L42, since it is now set upon tracker.Start(...), and not from 'AddSpans'. tracker.Start(...) is called from exporter/signalfxexporter/factory.go L94